ambry-network/src/main/java/com.github.ambry.network/BlockingChannelConnectionPool.java

```diff
@@ -44,16 +44,18 @@
   private final String host;
   private final Port port;
   private final Logger logger = LoggerFactory.getLogger(getClass());
-  private Gauge<Integer> availableConnections;
-  private Gauge<Integer> activeConnections;
-  private Gauge<Integer> totalNumberOfConnections;
+  protected Gauge<Integer> availableConnections;
```

Comment:
Yes. We don't have a good option either. 1. Expose the actual data structures (linkedin blocking queue) for active and available connections count. 2. Expose the gauge as we have done now. 3. Inherit the class and define methods to expose the sizes. But even this might involve changing the access specifier of the queues as protected